earthfile2llb/listener.go

```diff
@@ -595,7 +595,12 @@ func (l *listener) EnterEnvArgKey(c *parser.EnvArgKeyContext) {
 	if l.shouldSkip() {
 		return
 	}
-	l.envArgKey = c.GetText()
+	var envKey = c.GetText()
+	var matchedEnvKey = checkShellName(envKey)
+	if matchedEnvKey != envKey {
+		l.err = fmt.Errorf("Invalid env key definition %s", envKey)
```

Comment:
Best to add a return so that nothing else is applied.

```suggestion
		l.err = fmt.Errorf("Invalid env key definition %s", envKey)
		return
```